packages/x-charts/src/internals/plugins/featurePlugins/useChartCartesianAxis/useChartCartesianInteraction.selectors.ts

```diff
@@ -1,77 +1,165 @@
+);
+
+/**
+ * Get interaction values
+ */
+
+type Value = number | Date | null;
+
+function valueGetter(
+  value: number,
+  axes: ComputeResult<ChartsAxisProps>,
+  indexes: number,
+  ids?: AxisId,
+): Value;
+function valueGetter(
+  value: number,
+  axes: ComputeResult<ChartsAxisProps>,
+  indexes: number[],
+  ids: AxisId[],
+): Value[];
+function valueGetter(
+  value: number,
+  axes: ComputeResult<ChartsAxisProps>,
+  indexes: number | number[],
+  ids: AxisId | AxisId[] = axes.axisIds[0],
+): Value | Value[] {
+  return Array.isArray(ids)
+    ? ids.map((id, axisIndex) =>
+        getAxisValue(axes.axis[id], value, (indexes as number[])[axisIndex]),
+      )
+    : getAxisValue(axes.axis[ids], value, indexes as number);
+}
+export const selectorChartsInteractionXAxisValues = createSelector(
+  [
+    selectorChartsInteractionPointerX,
+    selectorChartXAxis,
+    selectorChartsInteractionXAxisIndexes,
+    optionalGetAxisIds,
+  ],
+  (value, axes, indexes, ids) => (value === null ? null : valueGetter(value, axes, indexes!, ids)),
+);
+
+export const selectorChartsInteractionXAxisValue = createSelector(
+  [
+    selectorChartsInteractionPointerX,
+    selectorChartXAxis,
+    selectorChartsInteractionXAxisIndex,
+    optionalGetAxisId,
+  ],
+  (x, xAxes, xIndex, id) => {
+    if (x === null || xIndex === null || xAxes.axisIds.length === 0) {
       return null;
     }
-    return getAxisIndex(yAxes.axis[yAxes.axisIds[0]], y);
+    return valueGetter(x, xAxes, xIndex, id);
   },
 );
 
+export const selectorChartsInteractionYAxisValues = createSelector(
+  [
+    selectorChartsInteractionPointerY,
+    selectorChartYAxis,
+    selectorChartsInteractionYAxisIndexes,
+    optionalGetAxisIds,
+  ],
+  (value, axes, indexes, ids) => (value === null ? null : valueGetter(value, axes, indexes!, ids)),
+);
+
 export const selectorChartsInteractionYAxisValue = createSelector(
-  [selectorChartsInteractionPointerY, selectorChartYAxis, selectorChartsInteractionYAxisIndex],
-  (y, yAxes, yIndex) => {
-    if (y === null || yIndex === null || !yAxes.axis[yAxes.axisIds[0]]) {
+  [
+    selectorChartsInteractionPointerY,
+    selectorChartYAxis,
+    selectorChartsInteractionYAxisIndex,
+    optionalGetAxisId,
+  ],
+  (y, yAxes, yIndex, id) => {
+    if (y === null || yIndex === null || yAxes.axisIds.length === 0) {
       return null;
     }
-    return getAxisValue(yAxes.axis[yAxes.axisIds[0]], y, yIndex);
+    return valueGetter(y, yAxes, yIndex, id);
   },
 );
 
-export type AxisValueIdentifier = {
-  value: number | Date | string;
-  // Set to -1 if no index.
-  index: number;
-};
-
-// TODO: probably remove in favor of the two more specific.
-export const selectorChartsInteractionXAxis = createSelector(
-  [selectorChartsInteractionXAxisIndex, selectorChartsInteractionXAxisValue],
-  (index, value): AxisValueIdentifier | null =>
-    index === null || value === null ? null : { index, value },
-);
+/**
+ * Know if tooltip is active for y-axis
```

Comment:
Is this comment accurate? It's preceding a selector called `selectorChartsInteractionTooltipXAxes`, which leads me to believe that something is off.